### PR TITLE
fix(non_annotated_t4_to_deepen): fix frame skipping logic

### PR DESCRIPTION
**Pull request by miursh**
## Description

Fixes a bug in the frame skipping logic that was introduced in #114, where frames were being skipped incorrectly under certain conditions. 
This fix ensures that frame skipping now functions as intended

### CODE REVIEW

File: perception_dataset/deepen/non_annotated_t4_to_deepen_converter.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ def _convert_one_frame(self, input_dir: str, output_dir: str, frame_index: int):
 
         for camera_sensor_type in self._camera_sensor_types:
             camera_channel = camera_sensor_type.value["channel"]
+            camera_token = None
```
Comment (kminoda) [inline perception_dataset/deepen/non_annotated_t4_to_deepen_converter.py:102]: Can we isolate this part as a separate function? Maybe we can add a unit test later on.

Function would look like e.g.
```python
def get_camera_token(camera_channel: str, sample, nusc) -> str | None:
    """Get camera token for `camera_channel` in the given `sample` data from a NuScenes dataset.
    Args:
        camera_channel (str): Camera channel name e.g. CAMERA_FRONT to look for
        sample: Sample data for a specific frame from NuScenes = `nusc.sample[frame_index]`
        nusc (NuScenes): NuScenes dataset
    Return:
        camera_token: str | None
            None if not found
    """
    camera_token: str | None = None
    if camera_channel in sample["data"].keys():
        camera_token = sample["data"][camera_channel]
    else:
        sample_data = [s for s in nusc.sample_data if s["sample_token"] == sample["token"]]
        for sensor in sample_data:
            if sensor["channel"] == camera_channel:
                camera_token = sensor["token"]
                break
    return camera_token
```

And then this part would be modified as
```python
            camera_token: str | None = get_camera_token(camera_channel, sample, nusc)
```

**Review by kminoda — COMMENTED** [review]
Thank you for the fix :pray:

File: perception_dataset/deepen/non_annotated_t4_to_deepen_converter.py (hunk 1 above)
Comment (miursh) [reply perception_dataset/deepen/non_annotated_t4_to_deepen_converter.py]: @kminoda 
Thanks for your advice.
https://github.com/tier4/tier4_perception_dataset/pull/153/commits/4fe5f64c44b6368dbe2f1fa7fe7f08d5cf9ec941

**Comment by miursh** [comment]
@kminoda 
Could you review this again?

**Review by kminoda — APPROVED** [review]

**Pull request merged**